src/bp/core/routers/bots/content.ts

```diff
@@ -129,6 +134,70 @@ export class ContentRouter extends CustomRouter {
         res.sendStatus(200)
       })
     )
+
+    this.router.get(
+      '/export',
+      this._checkTokenHeader,
+      this._needPermissions('read', 'bot.content'),
+      async (req, res) => {
+        const elements = await this.cms.getAllElements(req.params.botId)
+        const filtered = elements.map(x => _.omit(x, ['createdBy', 'createdOn', 'modifiedOn']))
+
+        res.setHeader('Content-Type', 'application/json')
+        res.setHeader('Content-disposition', `attachment; filename=content_${moment().format('DD-MM-YYYY')}.json`)
+        res.end(JSON.stringify(filtered, undefined, 2))
+      }
+    )
+
+    const upload = multer()
+    this.router.post(
+      '/analyzeImport',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      upload.single('file'),
+      this.asyncMiddleware(async (req: any, res) => {
+        try {
+          const existingElements = await this.cms.getAllElements(req.params.botId)
+          const contentTypes = (await this.cms.getAllContentTypes(req.params.botId)).map(x => x.id)
+
+          const importData = (await validate(JSON.parse(req.file.buffer), CmsImportSchema)) as ContentElement[]
+          const importedContentTypes = _.uniq(importData.map(x => x.contentType))
+
+          res.send({
+            cmsCount: (existingElements && existingElements.length) || 0,
+            fileCmsCount: (importData && importData.length) || 0,
+            missingContentTypes: _.difference(importedContentTypes, contentTypes)
```

Comment:
missing, is missing content elements to override all elements ?

Reply:
This makes sure that every content types used by the imported elements are present, otherwise they must be added beforehand.